### ISSUE TITLE
Passing empty splatted keyword args fails unless at least one keyword arg is declared

### ISSUE THREAD

**Issue by dazuma** [oracle/truffleruby#2567]
```ruby
def foo
  puts "Succeeded!"
end
kwargs = {}
foo(**kwargs)
```

This succeeds on MRI (2.6.9, 2.7.5, and 3.1.0 tested).
On TruffleRuby 21.3.0, it fails with `wrong number of arguments (given 1, expected 0) (ArgumentError)`.

Note: A similar issue was present in MRI prior to 2.7.0. MRI 2.6.x and earlier would give the same ArgumentError *only* on initialize methods. For example:

```ruby
class Foo
  def initialize
    puts "Succeeded!"
  end
end
kwargs = {}
Foo.new(**kwargs)
```

This fails on MRI 2.6.9, but succeeds on MRI 2.7.5 and 3.1.0. (It also fails on TruffleRuby 21.3.0.)

**Comment by eregon** [comment]
Thanks for report.

Interesting, I thought `foo(**empty_kwargs)` would definitely pass a positional empty Hash on 2.6.
It is the general semantics on <= 2.6, so this seems a weird corner case.
Maybe it's even a bug to not pass anything, because in many other cases 2.6 would pass an empty Hash in that case (which can be received as positional or as keywords with `**kw` on the callee side).

In general it's better to avoid relying on caller-side `**empty` passng something or not, and instead use `ruby2_keywords`: https://eregon.me/blog/2021/02/13/correct-delegation-in-ruby-2-27-3.html

In any case, we will fix this for 22.1.0 when we'll have Ruby 3 keyword argument handling (being worked on currently).

**Comment by eregon** [comment]
Fixed in https://github.com/oracle/truffleruby/commit/b8086afe6ecc64a78e1f80f42d4ae9c05cf56605
